### ISSUE TITLE
Сборка NDK

### ISSUE THREAD

**Issue by clclon** [AGENTxXx/improve-screencap-for-android#1]
С удовольствием прочёл вашу статью на Хабре к которой относится этот репозиторий и немного был смущён к подходу сборки. Рекомендую рассмотреть варианты изготовления библиотек заглушка для сборки, это не так сложно как кажется, но вносит удобства сборки под любую версию платформы Андроид и так-же архитектуру ЦПУ. Вот например: https://github.com/ClnViewer/Android-fast-screen-capture/tree/master/fakelib

Собирать с помощью NDK можно из ide: https://github.com/ClnViewer/Code-Blocks-Android-NDK
например)

**Comment by AGENTxXx** [comment]
> С удовольствием прочёл вашу статью на Хабре к которой относится этот репозиторий и немного был смущён к подходу сборки. Рекомендую рассмотреть варианты изготовления библиотек заглушка для сборки, это не так сложно как кажется, но вносит удобства сборки под любую версию платформы Андроид и так-же архитектуру ЦПУ. Вот например: https://github.com/ClnViewer/Android-fast-screen-capture/tree/master/fakelib
> 
> Собирать с помощью NDK можно из ide: https://github.com/ClnViewer/Code-Blocks-Android-NDK
> например)

Благодарю за информацию! Хорошо, я еще раз гляну. Вы сами пробовали данный способ? Я пробовал через fakelib и да, сборка создавалась, но потом, когда я переносил на устройство при запуске выдавалась ошибка, что-то типа "неверный линк на библиотеку такую-то". Возможно надо было собирать с какими-то определенными параметрами, но этим подходом не удалось. Я все что можно было в сети тогда перерыл и попробовал (не на столько я извращенец, чтобы колотить свои велосипеды, когда есть решения). Но как вариант, гляну, возможно, что-то я все же упустил...

**Comment by clclon** [comment]
Да, конечно пробывал ) Вот более обнавленная версия fakelib с поддержкой android::Vector https://github.com/clclon/Android-fast-screen-capture/tree/master/fakelib для сборки приложения с fakelib нужны также оригинальные AOSP headers от той версии андроид под которую собираете.
Но в принципе это не сильно принципиально, я с хеадерами 6.1 собирал под 10 андроид.

Андроид headers можно взять например тут https://github.com/ClnViewer/android-platform-headers
а можно утащить и из Гита Гугла.
Собирать ndk_build.cmd скриптом из поставки NDK
